### ISSUE TITLE
Validator is blocking all upcoming ensures when using 'then()'

### ISSUE THREAD

**Issue by OzgurDogan** [aurelia/validation#393]
**I'm submitting a bug report**

* **Library Version:**
0.14.0

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
5.0.0

* **NPM Version:**
3.3.6

* **JSPM OR Webpack AND Version**
JSPM 0.16.48

* **Browser:**
all

* **Language:**
TypeScript 2.0

**Current behavior:**
There is a difference between the Validator and the ValidationController with sequencing of the rules.
As noted in the documentation

> Rules are evaluated in parallel. Use the .then() method to postpone evaluation of a rule until after the preceding rules in the ensure have been evaluated.

I've created exact the same examples using the ValidationController and the other the Validator. But the outcome is different in both cases. Because with the ValidationController the 'then()' is blocking the rules per ensure. While with the Validator the 'then()' is block all upcoming rules for all ensures.

With the Validator:
https://gist.run/?id=5fad92eb25dfc18e4243eccb47411645

With the ValidationController:
https://gist.run/?id=1d38aa036096a5f2a0a2fac58ec187fd

**Expected/desired behavior:**
That the Validator is also behaving the same as ValidationController and the 'then()' is only postponing the evaluation of the rules per ensure.

**Comment by EisenbergEffect** [comment]
Ping @jdanyow
